### DISCUSSION TITLE
'fd' as a prefix is needlessly confusing

### DISCUSSION THREAD

**Discussion by bji** [firedancer-io/firedancer#1931] [General]
'fd' has meant "file descriptor" in the Unix world since 1969.  Using that prefix in your code and tooling creates a *lot* of unnecessary confusion.

As the most recent example, I am working on running a Firedancer node on testnet and am quite certain that I will never parse "fdctl" as anything other than "file descriptor control" and every time I see the binary, have to remind myself of what it really is.

Also the code base is very confusing to read for this same reason.

I recommend a global sed replace of " fd" with " firedan" or similar.

**Comment by bji** [comment]
Actually even better than "firedan" would be "fv".  For "FireDancer Validator".  I did a check and there is literally no binary on my entire Linux system that starts with 'fv', so I expect the namespace is completely wide open.

**Comment by ripatel-fd** [comment]
Moved this to discussions.

I don't like the name fdctl to launch Firedancer either. It is cryptic and doesn't make sense. "fdctl" literally means "file descriptor control" to any sysadmin. (Another issue is that the *ctl pattern implies there is some daemon it controls, but fdctl itself is the daemon) This binary should just be called `firedancer`. Open to renaming all the other binaries.

> I recommend a global sed replace of " fd" with " firedan" or similar

No, absolutely not. There is no discussion here about internal code identifiers. This project is too far along the development cycle to do anything tree-wide. I'm not going to tell 25 developers to stop writing code for 3 days and rebase all their patches and bug fallout resulting from such a thing.

> I am working on running a Firedancer node on testnet

Great to hear. Let us know how it goes (probably still too buggy)

**Comment by bji** [comment]
Yeah I guess I should have said something when I first read the code base a year ago and got confused with the fd_ everywhere.  fv_ would be much better but I guess it's too late now.  Thanks for at least considering renaming fdctl.
